### DISCUSSION TITLE
creating elaborate window decorations using titlebars and gears.shape

### DISCUSSION THREAD

**Discussion by ghost** [awesomeWM/awesome#3827] [Q&A]
I'm hoping to eventually create the look of a more "3D" window decoration, and from what I've gathered so far, the process would be "draw 4 titlebars, and draw shapes on those titlebars as their background". 
(Here's an example of what I mean when I say 3D- there's a shadow behind windows, and the borders use different shades of grey for shading)
![example image](https://github.com/awesomeWM/awesome/assets/60542090/988c1eeb-d9cb-41e6-a687-8f16bdb33f41)

buuut I've never really done anything like this before, and i've been struggling to make sense of the documentation. I would really appreciate some examples, even if its just "heres how you could check the client dimensions & draw a rectangle that wide"

**Comment by actionless** [comment]
you just define a template for the titlebar in a same way as for any other widget, here's my particular example for such sort of shadows:

https://github.com/actionless/awesome_config/blob/master/actionless/titlebar.lua#L580

![](https://camo.githubusercontent.com/94829c3d0a8904cc74205865fdb929d30766dc82fa802cae7591d3ba7269ccde/68747470733a2f2f692e726564642e69742f6872653874783976796e79782e706e67)

i also remember @Elv13 was posting a link to much more compact examples of similar retro titlebars
